### PR TITLE
remove optional symbols

### PR DESCRIPTION
**Pull request by bratpiorka**
related to https://github.com/oneapi-src/unified-memory-framework/issues/568

### CODE REVIEW

File: src/unsupported.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <umf.h>
+#include <umf/memspace.h>
+
+#if UMF_NO_HWLOC
+
+umf_result_t umfMemspaceCreateFromNumaArray(unsigned *nodeIds, size_t numIds,
```
Comment (lukaszstolarczuk) [inline src/unsupported.c:15]: just thinking out loud - should we have a test for that...?

Comment (bratpiorka) [reply src/unsupported.c:15]: We have jobs in our CI that use UMF_DISABLE_HWLOC and use this code.

Comment (lukaszstolarczuk) [reply src/unsupported.c:15]: hmm... I'm just saying, this precise function is never called in our tests; we enable memspace numa tests only on Linux when hwloc is enabled - code cov would confirm this 😄 

anyway, I think if this compiles it should be enough, to fix the issue, I guess...

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -151,14 +151,17 @@ if(NOT UMF_DISABLE_HWLOC)
...
-        add_optional_symbol(umfMemspaceCreateFromNumaArray)
         add_optional_symbol(umfMemspaceHighestBandwidthGet)
         add_optional_symbol(umfMemspaceHighestCapacityGet)
         add_optional_symbol(umfMemspaceHostAllGet)
         add_optional_symbol(umfMemspaceLowestLatencyGet)
     endif()
 endif()
 
+if(UMF_DISABLE_HWLOC OR WINDOWS)
+    set(UMF_SOURCES ${UMF_SOURCES} memspaces/memspace_unsupported_numa.c)
```
Comment (lplewa) [inline src/CMakeLists.txt:162]: memspace_numa_unsupported.c 

It matter as files are sorted alphabetically, so you we have both versions of the file next to each other.

File: src/memspaces/memspace_unsupported_numa.c [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/*
+ *
+ * Copyright (C) 2023-2024 Intel Corporation
```
Comment (lplewa) [inline src/memspaces/memspace_unsupported_numa.c:3]: 2024

File: src/CMakeLists.txt (hunk 2 above)
Comment (bratpiorka) [reply src/CMakeLists.txt]: done

File: src/memspaces/memspace_unsupported_numa.c (hunk 3 above)
Comment (bratpiorka) [reply src/memspaces/memspace_unsupported_numa.c]: done

**Review by lplewa — APPROVED** [review]

**Review by ldorau — APPROVED** [review]

**Comment by lukaszstolarczuk** [comment]
@bratpiorka when the CI is green should we merge this or is it a draft for a reason?

**Comment by bratpiorka** [comment]
> @bratpiorka when the CI is green should we merge this or is it a draft for a reason?

I'm still working on this

File: CMakeLists.txt [hunk 4]
```diff
@@ -423,29 +423,6 @@ if((UMF_BUILD_GPU_TESTS OR UMF_BUILD_GPU_EXAMPLES) AND UMF_BUILD_CUDA_PROVIDER)
     # TODO do the same for ze_loader
```
Comment (lukaszstolarczuk) [inline CMakeLists.txt:423]: you could add a sentence or two in the commit msg

File: CMakeLists.txt [hunk 5]
```diff
@@ -423,29 +423,6 @@ if((UMF_BUILD_GPU_TESTS OR UMF_BUILD_GPU_EXAMPLES) AND UMF_BUILD_CUDA_PROVIDER)
...
 endif()
 
-# set optional symbols for map/def files
-#
-# TODO: ref. #649
```
Comment (lukaszstolarczuk) [inline CMakeLists.txt:428]: pls mark this issues (649) as fixed

**Review by lplewa — APPROVED** [review]

**Review by ldorau — APPROVED** [review]

**Pull request merged**